### PR TITLE
Pundit policies

### PR DESCRIPTION
**Pull request by mcolombo87**
### Summary
New Policy for Users' Rent. An authenticated user only can see his rents, if he try to get any other an 403 code will be shoot it.
### ScreenShots
image
### Trello
https://trello.com/c/sin69Z80/12-instalar-y-configurar-pundit

### CODE REVIEW

File: app/controllers/application_controller.rb [hunk 1]
```diff
@@ -1,6 +1,15 @@
...
+  include Pundit
   # For APIs, you may want to use :null_session instead.
   protect_from_forgery with: :null_session
+  rescue_from Pundit::NotAuthorizedError, with: :user_not_authorized
+
+  private
+
+  def user_not_authorized
+    error_message = 'You are not authorized to perform this action.'
+    render json: { error: error_message }, status: :forbidden
```
Comment (facucompe) [inline app/controllers/application_controller.rb:13]: There is an `:unauthorized` status

File: app/controllers/api/v1/rents_controller.rb [hunk 2]
```diff
@@ -4,12 +4,14 @@ class RentsController < ApplicationController
       before_action :authenticate_user!
       def index
         user = User.find(params[:user_id])
+        authorize user.rents[0]
```
Comment (facucompe) [inline app/controllers/api/v1/rents_controller.rb]: You are just authorizing the first one

**Review by facucompe — CHANGES_REQUESTED** [review]

File: app/controllers/application_controller.rb (hunk 1 above)
Comment (mcolombo87) [reply app/controllers/application_controller.rb:13]: Yes, but maybe that be confused if the user is authenticated but have restricted access to this information. Forbidden it's best choose for me. 

> 403 Code: "The server understood the request but refuses to authorize it" (https://httpstatuses.com/403)

File: app/controllers/api/v1/rents_controller.rb (hunk 2 above)
Comment (mcolombo87) [reply app/controllers/api/v1/rents_controller.rb]: Yes, always will be rents for this user_id, so i not need to check all. Check one is enough.

Comment (facucompe) [reply app/controllers/api/v1/rents_controller.rb]: hahah, don't check at all then, usually we only check creation

Comment (floresv) [reply app/controllers/api/v1/rents_controller.rb]: mmm.. I think you should see only your rents. Apply a select or filter ?

**Review by floresv — COMMENTED** [review]
Review it please, Maybe a show of rents ? :thinking:

**Comment by mcolombo87** [comment]
@floresv "mmm.. I think you should see only your rents. Apply a select or filter ?" User only see his rents, always. Pundit take care of that.

File: app/controllers/api/v1/rents_controller.rb (hunk 2 above)
Comment (mcolombo87) [reply app/controllers/api/v1/rents_controller.rb]: Done!

**Review by facucompe — APPROVED** [review]

**Pull request merged**